### DISCUSSION TITLE
How to use thermal model of GEM5

### DISCUSSION THREAD

**Discussion by shikun-wei** [gem5/gem5#1315] [Q&A]
I'm trying to use power and thermal model https://www.gem5.org/documentation/general_docs/thermal_model to study the power consumption of processor. I have modified the file se.py by adding the thermal model. The code is like: 
thermal_node1 = ThermalNode()
thermal_node2 = ThermalNode()

# Create thermal domain and add nodes
thermal_domain = ThermalDomain()
thermal_domain.nodes = [thermal_node1, thermal_node2]

# Create thermal resistors and capacitors
thermal_resistor = ThermalResistor(node1=thermal_node1, node2=thermal_node2, resistance=0.5)
thermal_capacitor = ThermalCapacitor(node=thermal_node1, capacitance=100)

# Create thermal model and add components
thermal_model = ThermalModel()
thermal_model.domains = [thermal_domain]
thermal_model.resistors = [thermal_resistor]
thermal_model.capacitors = [thermal_capacitor]

# Attach the thermal model to the system
system.thermal_model = thermal_model
Then when I test it on helloworld application, I encounter a Segmentation fault and it seems that it doesn't exist a tutorial or an example script to show how to use the thermal model of GEM5. Does anyone have used it successfully and could tell me how to integrate it into simulation?

**Comment by saurabh-singh-rajput** [comment]
hey did you find any fix the this? going through the same issue, not many resources for the thermal model available?
